topup/keeper.go

```diff
@@ -0,0 +1,129 @@
+package topup
+
+import (
+	"math/big"
+	"strconv"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/maticnetwork/heimdall/bank"
+	"github.com/maticnetwork/heimdall/staking"
+	"github.com/maticnetwork/heimdall/topup/types"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+var (
+	// DefaultValue default value
+	DefaultValue = []byte{0x01}
+	// TopupSequencePrefixKey represents topup sequence prefix key
+	TopupSequencePrefixKey = []byte{0x81}
+)
+
+// ModuleCommunicator manager to access validator info
+type ModuleCommunicator interface {
```

Comment:
You don't need a module communicator since you have a staking keeper in the topup keeper.